tools/c7n_gcp/tests/test_sql.py

```diff
@@ -0,0 +1,71 @@
+# Copyright 2018 Capital One Services, LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import time
+from gcp_common import BaseTest
+
+
+class SqlInstanceTest(BaseTest):
+
+    def test_sqlinstance_query(self):
+        factory = self.replay_flight_data('sqlinstance-query')
+        p = self.load_policy(
+            {'name': 'all-sqlinstances',
+             'resource': 'gcp.sql-instance'},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+
+    def test_sqlinstance_get(self):
+        factory = self.replay_flight_data('sqlinstance-get')
+        p = self.load_policy(
+            {'name': 'one-sqlinstance',
+             'resource': 'gcp.sql-instance'},
+            session_factory=factory)
+        instance = p.resource_manager.get_resource(
+            {"project": "cloud-custodian",
+             "name": "brenttest-2"})
+        self.assertEqual(instance['state'], 'RUNNABLE')
+
+    def test_stop_instance(self):
+        project_id = 'cloud-custodian'
+        factory = self.replay_flight_data('sqlinstance-stop', project_id=project_id)
+        p = self.load_policy(
+            {'name': 'istop',
+             'resource': 'gcp.sql-instance',
+             'filters': [{'name': 'brenttest-5'}],
+             'actions': ['stop']},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        if self.recording:
+            time.sleep(1)
+
+    def test_delete_instance(self):
+        project_id = 'cloud-custodian'
+        factory = self.replay_flight_data('sqlinstance-terminate', project_id=project_id)
+        p = self.load_policy(
+            {'name': 'iterm',
+             'resource': 'gcp.sql-instance',
+             'filters': [{'name': 'brenttest-4'}],
+             'actions': ['delete']},
+            session_factory=factory)
+        resources = p.run()
+        self.assertEqual(len(resources), 1)
+        if self.recording:
+            time.sleep(1)
+        client = p.resource_manager.get_client()
+        result = client.execute_query(
+            'list', {'project': project_id})
+        self.assertEqual(len(result['items']), 2)
```

Comment:
this test should be retrieving the named instance to verify its status is now deleting. as is the test assertion list query is missing any assertion of the behavior of the delete action, as well picking/up recording an instance unrelated to the behavior being tested.

Reply:
@kapilt I remember now why I did it this way. When you issue the delete request,it immediately removes it and it's no longer available via the api to query. I, like you, thought it would mark it with a status of "pending_delete" or something similar but it doesn't.

My thought for this test was to do a before and after count. I.E., Count how many resources come back via a list before the delete, do the delete, then do another count and if the count is 1 less than before, the test passes. 

Any other way that you can think of to make sure this test passes to your liking?

Reply:
if you query it, and its not there, thats a valid assertion of the delete.

ie. in pseudo logic

```
try:
   instance = client.execute_query('get')
   self.fail('found deleted instance: %s' % instance)
except ClientError as e:
   self.assertTrue("not found" in str(e))
```

Reply:
Commited code based upon your recommendation for the test.